### PR TITLE
feat(apps/interface): implement 404 Page

### PR DESCRIPTION
**Pull request by math-marcellino**
Implement 404 page

### CODE REVIEW

File: apps/interface/pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    <Text
+                        fontSize="16px"
+                        lineHeight="24px"
+                        align="center"
+                        color={gray10}
+                    >
+                        Uh oh, seems the page you're searching is not there
+                    </Text>
+                </VStack>
+                <a href="https://risedle.com/">
```
Comment (pyk) [inline apps/interface/pages/404.tsx:63]: @math-marcellino use baseUrl instead of this

```
const baseConfig = getBaseConfig();

baseConfig.baseURL
```

File: apps/interface/pages/404.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    <Button variant="gradient">
+                        <Text marginRight="8px">&larr;</Text>
+                        Back to Landing Page
```
Comment (pyk) [inline apps/interface/pages/404.tsx:66]: s/Landing Page/Home/

**Review by pyk — CHANGES_REQUESTED** [review]
Just need smol update on the URL and link text

**Comment by pyk** [comment]
@math-marcellino also we can increase code coverage here:

```
interface:test: ---------------------------|---------|----------|---------|---------|-------------------
interface:test: File                       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
interface:test: ---------------------------|---------|----------|---------|---------|-------------------
interface:test: All files                  |    95.8 |      100 |   96.42 |   95.48 |                   
interface:test:  components/ChainIcon      |     100 |      100 |     100 |     100 |                   
interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:  components/DarkmodeToggle |     100 |      100 |     100 |     100 |                   
interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:  components/NavigationBar  |     100 |      100 |     100 |     100 |                   
interface:test:   chain.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:   link.tsx                 |     100 |      100 |     100 |     100 |                   
interface:test:   links.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:   logo.tsx                 |     100 |      100 |     100 |     100 |                   
interface:test:  components/WarningBar     |     100 |      100 |     100 |     100 |                   
interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:  hooks                     |     100 |      100 |     100 |     100 |                   
interface:test:   useViewportSize.ts       |     100 |      100 |     100 |     100 |                   
interface:test:  pages                     |      70 |      100 |   66.66 |      70 |                   
interface:test:   404.tsx                  |       0 |      100 |       0 |       0 | 8-14              
interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
interface:test:  utils                     |     100 |      100 |     100 |     100 |                   
interface:test:   getBaseConfig.ts         |     100 |      100 |     100 |     100 |                   
interface:test: ---------------------------|---------|----------|---------|---------|-------------------
```

add new spec in `tests/pages/404/index.tsx` make sure `<Custom404 />` render link correctly based on `getBaseConfig`. See example [here](https://github.com/risedle/monorepo/blob/main/apps/interface/tests/components/ChainIcon.spec.tsx#L9).

File: apps/interface/pages/404.tsx (hunk 1 above)
Comment (math-marcellino) [reply apps/interface/pages/404.tsx:63]: Got it sir, will fix it real quick..

File: apps/interface/pages/404.tsx (hunk 2 above)
Comment (math-marcellino) [reply apps/interface/pages/404.tsx:66]: Umm btw I don't quite understand about this one sir .-.

Comment (pyk) [reply apps/interface/pages/404.tsx]: @math-marcellino sorry `s/a/b` min replace string `a` with `b`

File: apps/interface/pages/404.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    <Text
+                        fontSize="16px"
+                        lineHeight="24px"
+                        align="center"
+                        color={gray10}
+                    >
+                        Uh oh, seems the page you're searching is not there
+                    </Text>
+                </VStack>
+                <a href={baseConfig.baseURL}>
```
Comment (pyk) [inline apps/interface/pages/404.tsx:65]: @math-marcellino sorry mathew, for link I think we can just use `/` instead of `baseURL`

- We want to do some prefetching too by using `NextLink`.
- It's recommended to use `Button` directly instead of wrapping it with `a`

For example:

```typescript
import NextLink from 'next/link'

<NextLink href='/' passHref>
  <Button as="a">Back to Home</Button>
</NextLink>
```

**Review by pyk — CHANGES_REQUESTED** [review]

File: apps/interface/pages/404.tsx (hunk 3 above)
Comment (math-marcellino) [reply apps/interface/pages/404.tsx:65]: Got it sir, let me fix this again 👍

**Comment by math-marcellino** [comment]
> @math-marcellino also we can increase code coverage here:
> 
> ```
> interface:test: ---------------------------|---------|----------|---------|---------|-------------------
> interface:test: File                       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
> interface:test: ---------------------------|---------|----------|---------|---------|-------------------
> interface:test: All files                  |    95.8 |      100 |   96.42 |   95.48 |                   
> interface:test:  components/ChainIcon      |     100 |      100 |     100 |     100 |                   
> interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:  components/DarkmodeToggle |     100 |      100 |     100 |     100 |                   
> interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:  components/NavigationBar  |     100 |      100 |     100 |     100 |                   
> interface:test:   chain.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:   link.tsx                 |     100 |      100 |     100 |     100 |                   
> interface:test:   links.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:   logo.tsx                 |     100 |      100 |     100 |     100 |                   
> interface:test:  components/WarningBar     |     100 |      100 |     100 |     100 |                   
> interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:  hooks                     |     100 |      100 |     100 |     100 |                   
> interface:test:   useViewportSize.ts       |     100 |      100 |     100 |     100 |                   
> interface:test:  pages                     |      70 |      100 |   66.66 |      70 |                   
> interface:test:   404.tsx                  |       0 |      100 |       0 |       0 | 8-14              
> interface:test:   index.tsx                |     100 |      100 |     100 |     100 |                   
> interface:test:  utils                     |     100 |      100 |     100 |     100 |                   
> interface:test:   getBaseConfig.ts         |     100 |      100 |     100 |     100 |                   
> interface:test: ---------------------------|---------|----------|---------|---------|-------------------
> ```
> 
> add new spec in `tests/pages/404/index.tsx` make sure `<Custom404 />` render link correctly based on `getBaseConfig`. See example [here](https://github.com/risedle/monorepo/blob/main/apps/interface/tests/components/ChainIcon.spec.tsx#L9).

Okay sir, let me familiarize myself with the testing first 👍

File: apps/interface/pages/404.tsx (hunk 3 above)
Comment (math-marcellino) [reply apps/interface/pages/404.tsx]: Btw @pyk , in here we changed the href from `baseConfig.baseURL` to `/`, so what should we write on the test instead(?) Btw sorry I'm quite confused here because I'm not used to write test for frontend :')

Comment (pyk) [reply apps/interface/pages/404.tsx]: @math-marcellino you can just make sure that link href is set to `/`.

here is how to test the `href` property https://github.com/risedle/monorepo/blob/0077bfd266c114b54097f4b19a71b3de84500ea0/apps/interface/tests/components/NavigationBar.spec.tsx#L41

File: apps/interface/pages/404.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    Text,
+    VStack,
+    useColorModeValue,
+} from "@chakra-ui/react";
+import NextLink from "next/link";
+import NavigationBar from "../components/NavigationBar";
+import { getBaseConfig } from "../utils/getBaseConfig";
+
+export default function Custom404() {
+    const baseConfig = getBaseConfig();
```
Comment (pyk) [inline apps/interface/pages/404.tsx:14]: @math-marcellino remove unused function and import

**Review by pyk — CHANGES_REQUESTED** [review]

File: apps/interface/tests/pages/404/index.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import "@testing-library/jest-dom/extend-expect";
+import nextRouter from "next/router";
+import { render, screen } from "@testing-library/react";
+import Custom404 from "../../../pages/404";
+
```
Comment (pyk) [inline apps/interface/tests/pages/404/index.spec.tsx:5]: Need to clean up the mock here

Example
https://github.com/risedle/monorepo/blob/2c25f874ece3b229cb6c9ed3b1b3a708ca0b88b0/apps/interface/tests/components/NavigationBar.spec.tsx#L6-L9

**Review by pyk — CHANGES_REQUESTED** [review]
Just need to clear the mock and we are good to go

File: apps/interface/tests/pages/404/index.spec.tsx (hunk 5 above)
Comment (math-marcellino) [reply apps/interface/tests/pages/404/index.spec.tsx:5]: Got it sir, it's fixed now 👍

File: apps/interface/pages/404.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                            404
+                        </Text>
+                    </VStack>
+                    <Text
+                        fontSize="16px"
+                        lineHeight="24px"
+                        align="center"
+                        color={gray10}
+                    >
+                        Uh oh, seems the page you're searching is not there
```
Comment (asaadam) [inline apps/interface/pages/404.tsx:61]: how about to change `'` to `&#39;`

File: apps/interface/pages/404.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+                direction="column"
+                grow={1}
+                gap="64px"
+                alignItems="center"
+                justifyContent="center"
+            >
+                <VStack spacing="32px">
+                    <VStack spacing="16px">
+                        <Text
+                            fontSize="12px"
```
Comment (asaadam) [inline apps/interface/pages/404.tsx:35]: matthew i think we can create theme for this text. 

i believe this will quite redundant in the future. 

further read you can read from here : https://chakra-ui.com/docs/styled-system/text-and-layer-styles

TLDR: create theme, then you can use like this


```
  return <Box textStyle='yourDefinedStyles'>This is a box</Box>

```

wdyt mas @pyk  ?

Comment (pyk) [reply apps/interface/pages/404.tsx:35]: @asaadam there are already built in value for this.

- `spacing="32px"` can be replaced with `spacing="8"`
- `spacing="16px"` can be replaced with `spacing="4"`
-  `fontSize="12px"` can be replaced with `fontSize="xs"`

@math-marcellino u can see list of default values here https://chakra-ui.com/docs/styled-system/theme#typography

**Review by pyk — APPROVED** [review]
